.github/workflows/convert.yml

```diff
@@ -0,0 +1,56 @@
+name: convert precommit and test
+run-name: convert precommit and test
+on:
+  pull_request:
+    paths:
+      - convert/**
+      - .github/workflows/convert.yml
+jobs:
+  convert-pre-commit-actions:
+    strategy:
+      matrix:
+        python-version: [ "3.8.15" ]
```

Comment:
Base image uses 3.8.10.
I believe your environment for tests and linters should be consistent with that.